Sources/SwiftCompilerPlugin/CompilerPlugin.swift

```diff
@@ -112,12 +112,45 @@ extension CompilerPlugin {
     let connection = try StandardIOMessageConnection()
     let provider = MacroProviderAdapter(plugin: Self())
     let impl = CompilerPluginMessageListener(connection: connection, provider: provider)
+    #if os(WASI)
+    // Rather than blocking on read(), let the host tell us when there's data.
+    readabilityHandler = {
+      do {
+        _ = try impl.handleNextMessage()
+      } catch {
+        internalError("\(error)")
+      }
+    }
+    #else
```

Comment:
I _feel_ this logic should be sinked to `impl.main()`, so `handleNextMessage()` can be `internal`.

Reply:
hmm I'm worried that moving `readabilityHandler` into `SwiftCompilerPluginMessageHandling` would be leaky, since the `swift_wasm_macro_pump` export (which invokes `readabilityHandler`) is an impl detail of the `SwiftCompilerPlugin` target. Also worth mentioning that `CompilerPluginMessageListener` is `@_spi(PluginMessage)` so I don't think we're creating any new API contracts even if `handleNextMessage` is technically public.

Reply:
Actually I just realized that `_SwiftSyntaxCShims` is a dependency of `SwiftCompilerPluginMessageHandling` so the latter is already strongly coupled to our wasm ABI. Moved `readabilityHandler` into that module.